galter_subjects_utils/keeptrace.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 Northwestern University.
+#
+# galter-subjects-utils is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""KeepTrace."""
+
+from dataclasses import dataclass
+
+
+@dataclass
+class KeepTrace:
```

Comment:
Now that keep trace related functionality is used outside of the updater, it made sense to extract it out in its own file.